src/main/kotlin/controller/LottoController.kt

```diff
@@ -1,51 +1,104 @@
+import model.LottoNumber
+import model.Lottos
+import model.ManualLottoPurchaseCount
+import model.NumberGenerator
+import model.Rank
+import model.WinningLotto
+import model.WinningStatistic
+import model.WinningStatistics
+import util.LottoConstants
 import view.InputView
 import view.OutputView
 
-class LottoController {
-    private lateinit var buyer: Buyer
-    private lateinit var winningStatistics: WinningStatistics
-
+class LottoController(private var gameState: GameState = GameState.Play) {
     fun run() {
-        val purchaseAmount = InputView.inputPurchaseAmount()
-        buyer = Buyer(purchaseAmount)
-        val lottos = publishLottos()
-        val winningLotto = drawWinningLotto()
-        winningStatistics = makeWinningStatics(lottos, winningLotto)
-        displayWinningStatistics(purchaseAmount)
+        while (true) {
+            when (gameState) {
+                GameState.Play -> play()
+                GameState.End -> break
+            }
+        }
+    }
+
+    private fun play() {
+        playLotto().onSuccess {
+            gameState = GameState.End
+        }.onFailure { e ->
+            println(e.message)
+        }
     }
 
-    private fun generateLottoNumbers(): List<Int> {
+    private fun playLotto() =
+        runCatching {
+            val purchaseAmount = InputView.inputPurchaseAmount()
+            val money = Buyer.from(purchaseAmount)
+            val purchaseSizeOfManualLotto = InputView.inputPurchaseSizeOfManualLotto()
+            val manualLottoCount = ManualLottoPurchaseCount.from(purchaseSizeOfManualLotto, money.numberOfLotto)
+            val userLotto = publishLottos(manualLottoCount, money)
+            val winningLotto = drawWinningLotto()
+
+            val winningStatistics = makeWinningStatics(userLotto, winningLotto)
+            displayWinningStatistics(money.numberOfLotto, winningStatistics)
+        }
+
+    private fun generateLottoNumbers(): List<LottoNumber> {
         val numberGenerator = NumberGenerator()
         val randomNumbers = numberGenerator.makeRandomNumbers()
 
-        return numberGenerator.drawSixNumbers(randomNumbers)
+        return numberGenerator.drawSixNumbers(randomNumbers).map { LottoNumber(it) }
     }
 
-    private fun publishLottos(): Lottos {
-        val lottos = Lottos(List(buyer.numberOfLotto) { Lotto(generateLottoNumbers()) })
-        buyer.buyLottos(lottos)
-        displayPurchaseResult()
-        return lottos
+    private fun publishLottos(
+        numberOfManualLotto: ManualLottoPurchaseCount,
+        money: Buyer,
+    ): Lottos {
+        val manual = InputView.inputManualLottos(numberOfManualLotto.count)
+        val manualLottos = Lottos(manual.map { generateManualLotto(it) })
+        val autoLottos = Lottos(List(money.numberOfLotto - manualLottos.publishedLottos.size) { generateAutoLotto() })
+
+        displayPurchaseResult(manualLottos, autoLottos)
+
+        return Lottos(manualLottos.publishedLottos + autoLottos.publishedLottos)
```

Comment:
코틀린에서는 정해진 이름의 함수를 오버로딩함으로써 표준 수학 연산자를 오버로딩할 수 있습니다. 아래의 글이 도움 될 거예요.
https://kotlinlang.org/docs/operator-overloading.html

```suggestion
        return manualLottos + autoLottos
```